### PR TITLE
Compatibility fixes

### PR DESCRIPTION
**Pull request by MartinMystikJonas**

### CODE REVIEW

File: Statement/Statement.php [hunk 1]
```diff
@@ -319,62 +319,32 @@ public function removeTransaction(TransactionInterface $transaction)
...
         return count($this->transactions);
     }
 
-    /**
-     * @see \Iterator::current()
-     *
-     * @return TransactionInterface
-     */
-    public function current()
+    public function current(): mixed
```
Comment (enumag) [inline Statement/Statement.php]: Refactor this class to implement IteratorAggregate instead of Iterator.

**Review by enumag — CHANGES_REQUESTED** [review]

File: Statement/Statement.php (hunk 1 above)
Comment (MartinMystikJonas) [reply Statement/Statement.php]: replaced

File: Statement/Statement.php [hunk 2]
```diff
@@ -319,63 +320,13 @@ public function removeTransaction(TransactionInterface $transaction)
...
-        return reset($this->transactions);
-    }
-
-    /**
-     * @see \Iterator::valid()
-     *
-     * @return bool
-     */
-    public function valid()
+    public function getIterator()
```
Comment (enumag) [inline Statement/Statement.php:332]:
```suggestion
    /**
     * @return Iterator<TransactionInterface>
     */
    public function getIterator(): Iterator
```

File: Statement/Statement.php [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
 namespace JakubZapletal\Component\BankStatement\Statement;
 
+use ArrayIterator;
```
Comment (enumag) [inline Statement/Statement.php:5]:
```suggestion
use ArrayIterator;
use Iterator;
```

**Review by enumag — CHANGES_REQUESTED** [review]

**Comment by MartinMystikJonas** [comment]
@enumag Could you merget his? Or is there anything else to do?

**Pull request merged**